### PR TITLE
Fix the error count for subscription requests for apollo telemetry

### PR DESCRIPTION
**Pull request by bnjjj**
Fix the error count for subscription requests for apollo telemetry

### CODE REVIEW

**Comment by BrynCooke** [comment]
Can we add a test for this?

File: .changesets/fix_bnjjj_fix_subscription_metrics.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+### Fix the error count for subscription requests for apollo telemetry ([PR #3500](https://github.com/apollographql/router/pull/3500))
+
+Fix the error count for subscription requests for apollo telemetry
```
Comment (o0Ignition0o) [inline .changesets/fix_bnjjj_fix_subscription_metrics.md:3]:
```suggestion
Count subscription requests only if the feature is enabled.

The router would previously count subscription requests regardless of whether the feature is enabled or not. This changeset will only count subscription requests if the feature has been enabled.
```

**Comment by bnjjj** [comment]
@BrynCooke test added

File: apollo-router/src/plugins/telemetry/metrics/snapshots/apollo_router__plugins__telemetry__metrics__apollo__test__apollo_metrics_for_subscription_error.snap [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            ],
+            "is_interface": false
+          }
+        }
+      }
+    },
+    "licensed_operation_count_by_type": {
+      "type": "subscription",
+      "subtype": "subscription-request",
+      "licensed_operation_count": 1
```
Comment (o0Ignition0o) [inline apollo-router/src/plugins/telemetry/metrics/snapshots/apollo_router__plugins__telemetry__metrics__apollo__test__apollo_metrics_for_subscription_error.snap:58]: I might have missunderstood, but i thought this number should be 0 🤔

Comment (bnjjj) [reply apollo-router/src/plugins/telemetry/metrics/snapshots/apollo_router__plugins__telemetry__metrics__apollo__test__apollo_metrics_for_subscription_error.snap:58]: No it's still counting the operation right now, but mark it with `has_errors` to `true`

**Review by o0Ignition0o — APPROVED** [review]

**Comment by o0Ignition0o** [comment]
@bnjjj do you mind creating a followup issue?

**Comment by bnjjj** [comment]
Thanks @o0Ignition0o https://github.com/apollographql/router/issues/3507

**Review by BrynCooke — APPROVED** [review]

**Pull request merged**